Meta OpInfo Test for stride correctness

SherlockNoMad opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #87849

Failing test logs here 
https://gist.github.com/SherlockNoMad/a7e132f3cb4152900f8a6d7df358c59e

SherlockNoMad committed:
Meta OpInfo Test for stride correctness

[ghstack-poisoned]

SherlockNoMad committed:
Update on "Meta OpInfo Test for stride correctness"


Failing test logs here 
https://gist.github.com/SherlockNoMad/a7e132f3cb4152900f8a6d7df358c59e 

[ghstack-poisoned]

SherlockNoMad committed:
Update on "Meta OpInfo Test for stride correctness"


Failing test logs here 
https://gist.github.com/SherlockNoMad/a7e132f3cb4152900f8a6d7df358c59e 

[ghstack-poisoned]

eellison commented on test/test_meta.py:
```diff
@@ -187,15 +188,79 @@ def test_tensor_outlives_converter(self):
-    torch.ops.aten.index_add,
+}
+
+CHECK_STRIDES_SKIPS = {
```
:'(

eellison commented on test/test_meta.py:
```diff
@@ -187,15 +188,79 @@ def test_tensor_outlives_converter(self):
+    aten._conj_physical.default,
+    aten._fft_c2c.default,
+    aten._fft_c2r.default,
+    aten._fft_r2c.default,
```
nit: we could make this expected failures by running it once with check_strides = True, asserting it fails if it's in this list, then running it again with check_strides=False

SherlockNoMad replied:
XFAIL in test harness is asserting for every sample... 
However, for the stride tests, not all sample will fail, so we still need to SKIP... :(

eellison commented on test/test_meta.py:
```diff
@@ -854,6 +917,40 @@ def __torch_function__(self, func, types, args=(), kwargs=None):
+            strided_arg_variants = [arg]
+        strided_args.append(strided_arg_variants)
+
+    for result in itertools.product(*strided_args):
```
we might want to check that we dont have any super long lists of tensor inputs for like `concat` or something to prevent exponential blowup. i.e., number of tensor args <= 4

SherlockNoMad replied:
limited to <=5, as batchnorm have 5 tensor inputs,

eellison reviewed (approved):
LGTM

SherlockNoMad committed:
Update on "Meta OpInfo Test for stride correctness"


Failing test logs here 
https://gist.github.com/SherlockNoMad/a7e132f3cb4152900f8a6d7df358c59e 

[ghstack-poisoned]

SherlockNoMad commented:
@pytorchbot merge -g
